frontend/src/components/placementVisuals/PlacementContainer.tsx

```diff
@@ -181,18 +181,43 @@ const PlacementContainer = ({ containerType, columns, rows, samples, direction,
         return id ? { ...samples[id], id: parseInt(id), type: type } : undefined
     }, [samples, selectedSampleList])
 
+    const letters = useMemo(() => "ABCDEFGHIJKLMNOPQRSTUVWXYZ".slice(0, rows), [rows])
+    const sampleValues = useMemo(() => Object.values(samples), [samples])
+
+    const selectColumn = useCallback((colNumber) => (e) => {
+        const mySample = [...letters].map((rowLetter) => {
+            const coordinate = rowLetter + "" + (padColumn(colNumber))
+            return sampleValues.find((sample: any) => (sample.coordinates) == coordinate)
+        }).filter(x => x)
+        updateSamples([...previewCells, ...mySample], containerType, rows, columns)
+    }, [sampleValues, columns, containerType, letters, padColumn, previewCells, rows, updateSamples])
+
+    const selectRow = useCallback((rowLetter) => (e) => {
+        const mySamples = [...Array(columns).keys()].map((c) => {
+            const colNumber = c + 1
+            const coordinate = rowLetter + "" + (padColumn(colNumber))
+            return sampleValues.find((sample: any) => (sample.coordinates) == coordinate)
+        }).filter(x => x)
+        updateSamples([...previewCells, ...mySamples], containerType, rows, columns)
+    }, [sampleValues, columns, containerType, padColumn, previewCells, rows, updateSamples])
+
+    const selectAll = useCallback((e) => {
+        updateSamples([...previewCells, ...sampleValues], containerType, rows, columns)
+    }, [sampleValues, updateSamples, previewCells, containerType, rows, columns])
+
     const renderCells = useCallback(
         () => {
             const cells: any[] = []
             let char = 'A'
             let coordinates = ''
             //renders header based on the number of columns provided to the component
             const headerCells: React.ReactElement[] = []
-            for (let i = 0; i < columns + 1; i++) {
+            const cellSize = columns <= 12 ? "cell" : "tiny-cell"
+	    for (let i = 0; i < columns + 1; i++) {
                 headerCells.push(
-                    <div key={'header_' + i} className={"header"}>
+                    <div key={'header_' + i} className={cellSize} style={{ backgroundColor: '#001529', color: 'white', cursor: 'grab' }} onClick={i !== 0 ? selectColumn(i) : selectAll}>
```

Comment:
Missing the stoppropagation() that prevent activating selection when clicking headers.

Reply:
there used to be a stoppropagation there actually, but i found that removing it works better
if i left stoppropagation there, it would not respond unless you click elsewhere to trigger selecting mode (edited)